### ISSUE TITLE
Default template not considered

### ISSUE THREAD

**Issue by stefanwalther** [update/updater-editorconfig#1]
The default template `~/templates/_editorconfig` is not being considered.

I use: 
- Node: 5.0.0
- Update: v0.6.2
- updater-config: v0.1.0

**Comment by jonschlinkert** [comment]
Can you try adding the template to `~/generate/templates`? Looks like the docs are wrong in a few places. sorry about that

**Comment by stefanwalther** [comment]
So you mean that also update is looking for templates in the ~/generate/templates folder?

To be honest, I really don't want to create one root folder for generate, another for generate, etc.?
Is this the plan?

**Comment by stefanwalther** [comment]
Doesn't matter which documentation I have a look into (generators or updaters), you always mention `~/templates/foo`

**Comment by jonschlinkert** [comment]
> So you mean that also update is looking for templates in the ~/generate/templates folder?

no, sorry typo, `update` will look in `~/update/templates`. But I just realized I haven't pushed up the code for update to use the custom directory yet anyway. 

> To be honest, I really don't want to create one root folder for generate, another for generate, etc.? Is this the plan?

Per my previous comments, I don't know lol. This is why I asked for your feedback. I personally don't want to create folders for each app either. I'd rather have one `~/templates` directory. But templates aren't the only thing you can customize with these apps, so I was leaning towards using `~/%appname%` as the root for user-defined config, templates etc. But I honestly don't care where stuff goes. 

Also, I mentioned this already but maybe you didn't see it. No matter what the defaults are, you will be _able to customize the paths to be whatever you want_ once I get a chance to implement the code.

**Comment by stefanwalther** [comment]
I double-checked it with `~/update/templates` and it doesn't work.

Another observation (will file that also to update since I believe it's a global issue):
- If I cd to a sub-directory, still the root directory of a project (actually where the package.json) will be considered, so in this case the editorconfig will be placed into the wrong directory.

Does this make sense?

**Comment by jonschlinkert** [comment]
> I haven't pushed up the code for update to use the custom directory yet

That's to be expected until I implement the feature.

> If I cd to a sub-directory, still the root directory of a project (actually where the package.json) will be considered, so in this case the editorconfig will be placed into the wrong directory.

with `generate` or `update`? It only cd's to the root of a project when:

- the directory is not empty, and
- the directory does not have a package.json

IMO this makes sense. Otherwise it would make it harder to do scaffolding in an existing project

**Comment by stefanwalther** [comment]
> If I cd to a sub-directory, still the root directory of a project (actually where the package.json) will be considered, so in this case the editorconfig will be placed into the wrong directory.

OK, I get it. In case of `.editorconfig`, `.gitattributes`, etc. this makes sense. But I think that's a limitation of the use-cases of update.

Just to give you an example:
- I am writing a lot of code-examples
- All these examples should be self-containing, so without any dependencies
- Therefore I'd like to go to directory `XYZ` and just do an `update add-my-magic-script` here
- I use this script in many examples, so ´update` is of big value for me
- But the `cd-up-to-package-json` is actually very limiting in my use-cases

Makes sense?
